api/src/main/scala/dcos/metronome/api/Authorization.scala

```diff
@@ -51,41 +51,26 @@ case class AuthorizedRequest[Body](identity: Identity, request: Request[Body], a
   def notAuthorized(): Result = PluginFacade.withResponse(authorizer.handleNotAuthorized(identity, _))
 }
 
-trait Authorization extends RestController {
-
-  def authenticator: Authenticator
-  def authorizer: Authorizer
-  def config: ApiConfig
-  def defaultBodyParser: BodyParser[AnyContent]
-  def metrics: Metrics
-
-  implicit val mat: Materializer
-
-  //play default execution context
-  import play.api.libs.concurrent.Execution.Implicits._
-
+abstract class Authorization(cc: ControllerComponents)(implicit ec: ExecutionContext, metrics: Metrics, authenticator: Authenticator, authorizer: Authorizer, config: ApiConfig, mat: Materializer) extends RestController(cc) {
   /**
     * Use this object to create an authorized action.
     */
-  object AuthorizedAction extends AuthorizedActionBuilder {
-    def apply() = new AuthorizedActionBuilder(None)
-    def apply(identity: Identity) = new AuthorizedActionBuilder(Some(identity))
-  }
+  def AuthorizedAction: ActionBuilder[AuthorizedRequest, AnyContent] =
+    Action andThen new AuthorizedActionBuilder()
+  def AuthorizedAction(identity: Identity): ActionBuilder[AuthorizedRequest, AnyContent] =
+    Action andThen new AuthorizedActionBuilder(Some(identity))
 
-  class AuthorizedActionBuilder(authorize: Option[Identity] = None) extends ActionBuilder[AuthorizedRequest, AnyContent] {
+  class AuthorizedActionBuilder(authorize: Option[Identity] = None) extends ActionRefiner[Request, AuthorizedRequest] {
+    override protected def executionContext: ExecutionContext = ec
 
-    def invokeBlock[A](request: Request[A], block: AuthorizedRequest[A] => Future[Result]) = {
+    override protected def refine[A](request: Request[A]): Future[Either[Result, AuthorizedRequest[A]]] = {
```

Comment:
Is this all changes needed for updated version of `Play`?

Reply:
This wasn’t necessary but getting rid of some of the warns it was easier to use ActionRefiner rather than the previous one. (It doesn’t require you to specify a bodyparser for example)